packages/sdk/lib/__tests__/unit.test.ts

```diff
@@ -0,0 +1,629 @@
+      );
+
+      const plugin = createPlugin();
+
+      const itly = new Itly();
+      itly.load({
+        context: {
+          a: 123,
+        },
+        plugins: [validationPlugin, plugin],
+        validation: Validation.ErrorOnInvalid,
+      });
+
+      expect(() => callItlyMethod(itly, methodName, methodArgs)).toThrow('Validation Error: context error');
+      expect(validationPlugin.validate).toHaveBeenCalledTimes(2);
+
+      const [pluginMethod, pluginPostMethod] = getPluginMethods(plugin, methodName);
+      expect(pluginMethod).toHaveBeenCalledTimes(0);
+      expect(pluginPostMethod).toHaveBeenCalledTimes(1);
+    });
+
+  test.each(callMethods.filter(([methodName]) => methodName !== 'track'))('should track valid event on %s if context is not valid and options.validation = ErrorOnInvalid',
+    (methodName, methodArgs) => {
+      const validationPlugin = createPlugin();
+      validationPlugin.validate = jest.fn(
+        (event: Event) => (event.name === 'context' ? { valid: false, message: 'context error' } : { valid: true }),
+      );
+
+      const plugin = createPlugin();
+
+      const itly = new Itly();
+      itly.load({
+        context: {
+          a: 123,
+        },
+        plugins: [validationPlugin, plugin],
+        validation: Validation.ErrorOnInvalid,
+      });
+
+      callItlyMethod(itly, methodName, methodArgs);
+      expect(validationPlugin.validate).toHaveBeenCalledTimes(1);
+
+      const [pluginMethod, pluginPostMethod] = getPluginMethods(plugin, methodName);
+      expect(pluginMethod).toHaveBeenCalledTimes(1);
+      expect(pluginPostMethod).toHaveBeenCalledTimes(1);
+    });
+});
+
+describe('plugin', () => {
+  const properties = { a: 'abc', b: 123 };
+
+  const trackMethods: [string, any[]][] = [
+    ['alias', ['user-1', 'user-2']],
+    ['identify', ['user-1', properties]],
+    ['group', ['user-1', 'group-1', properties]],
+    ['page', ['user-1', 'category', 'name', properties]],
+    ['track', ['user-1', { name: 'event-1', properties }]],
+  ];
+  const resetMethod: [string, any[]] = ['reset', []];
+  const flushMethod: [string, any[]] = ['flush', []];
+  const allMethods = [...trackMethods, resetMethod, flushMethod];
+
+  test.each(trackMethods)('should call %s if itly is not disabled', (methodName, methodArgs) => {
+    const plugin = createPlugin();
+
+    const itly = new Itly();
+    itly.load({
+      plugins: [plugin],
+    });
+
+    callItlyMethod(itly, methodName, methodArgs);
+
+    const [pluginMethod, pluginPostMethod] = getPluginMethods(plugin, methodName);
+    expect(pluginMethod).toHaveBeenCalledTimes(1);
+    if (methodName !== 'alias') {
+      expect(pluginPostMethod).toHaveBeenCalledTimes(1);
+    }
+    expect(pluginMethod.mock.calls[0].slice(0, -1)).toEqual(methodArgs);
+  });
+
+  test.each(trackMethods)('should not call %s if itly is disabled', (methodName, methodArgs) => {
```

Comment:
Should be allMethods here (+reset, flush), this is also a bug in the SDK.

Reply:
done